siddhi-test-framework/src/main/java/io/siddhi/distribution/test/framework/SiddhiRunnerContainer.java

```diff
@@ -209,18 +214,43 @@ private int getStartupTimeoutSeconds() {
     @Override
     protected void waitUntilContainerStarted() {
         logger().info("Waiting for Siddhi Runner Container to start...");
-        retryUntilSuccess(getStartupTimeoutSeconds(), TimeUnit.SECONDS, () -> {
-            if (!isRunning()) {
-                throw new ContainerLaunchException("Siddhi Runner Container failed to start");
-            }
-            HTTPClient.HTTPResponseMessage httpResponseMessage = callHealthAPI();
-            if (httpResponseMessage.getResponseCode() == 200) {
-                logger().info("Siddhi Runner Health API reached successfully.");
-                return null;
-            } else {
-                throw new ConnectException("Failed to connect with the Siddhi Runner health API");
+        if (deploymentDirectoryFile != null) {
+            String[] siddhiAppsArray = deploymentDirectoryFile.list();
+            assert siddhiAppsArray != null;
+            for (String siddhiApp : siddhiAppsArray) {
+                String fileName = siddhiApp.substring(0, siddhiApp.length() - 7);
+                retryUntilSuccess(getStartupTimeoutSeconds(), TimeUnit.SECONDS, () -> {
+                    if (!isRunning()) {
+                        throw new ContainerLaunchException("Siddhi Runner Container failed to start");
+                    }
+                    HTTPClient.HTTPResponseMessage httpResponseMessage = callHealthAPI();
+                    if (httpResponseMessage.getResponseCode() == 200) {
+                        String logs = this.getLogs();
+                        if (logs.contains("Siddhi App " + fileName + " deployed successfully")) {
+                            logger().info("Siddhi Runner Health API reached successfully.");
```

Comment:
Use proper messages. Have two info logs. One for the health api. Other for all apps deployed successfully